ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java

```diff
@@ -0,0 +1,1161 @@
+import io.confluent.ksql.logging.processing.ProcessingLogger;
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.serde.FormatFactory;
+import io.confluent.ksql.serde.FormatInfo;
+import io.confluent.ksql.serde.SerdeFeatures;
+import io.confluent.ksql.serde.StaticTopicSerde;
+import io.confluent.ksql.serde.WindowInfo;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Optional;
+import java.util.OptionalInt;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology.AutoOffsetReset;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.ValueMapper;
+import org.apache.kafka.streams.kstream.ValueTransformerWithKey;
+import org.apache.kafka.streams.kstream.ValueTransformerWithKeySupplier;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.SessionWindow;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class SourceBuilderV1Test {
+
+  private static final ColumnName K0 = ColumnName.of("k0");
+  private static final ColumnName K1 = ColumnName.of("k1");
+
+  private static final LogicalSchema SOURCE_SCHEMA = LogicalSchema.builder()
+      .keyColumn(K0, SqlTypes.BIGINT)
+      .valueColumn(ColumnName.of("field1"), SqlTypes.STRING)
+      .valueColumn(ColumnName.of("field2"), SqlTypes.BIGINT)
+      .build();
+
+  private static final LogicalSchema MULTI_COL_SOURCE_SCHEMA = LogicalSchema.builder()
+      .keyColumn(K0, SqlTypes.BIGINT)
+      .keyColumn(K1, SqlTypes.BIGINT)
+      .valueColumn(ColumnName.of("field1"), SqlTypes.STRING)
+      .valueColumn(ColumnName.of("field2"), SqlTypes.BIGINT)
+      .build();
+
+  private static final double A_KEY = 10.11;
+
+  private static final GenericKey KEY = GenericKey.genericKey(A_KEY);
+
+  private static final LogicalSchema SCHEMA_WITHOUT_NEW_PSEUDOCOLUMNS = SOURCE_SCHEMA
```

Comment:
Names like this don't age well -- once more pseudocolumns are added, what does "new pseducolumns" mean? Let's rename.

Reply:
Do you think `SCHEMA_WITH_LEGACY_PSEUDOCOLUMNS` would be good? Or, since it should never change, `SCHEMA_WITH_ONLY_ROWTIME_AS_PSEUDOCOLUMN`?

Reply:
I think something like `SCHEMA_WITH_V0_PSEUDOCOLUMNS`, `SCHEMA_WITH_ROWTIME`, or `SCHEMA_WITH_ROWTIME_AS_ONLY_PSEUDOCOLUMN` makes sense.